react/Popup/index.jsx

```diff
@@ -0,0 +1,199 @@
+import { PureComponent } from 'react'
+import PropTypes from 'prop-types'
+
+import { isMobileApp } from 'cozy-device-helper'
+
+/**
+ * Customized function to get dimensions and position for a centered
+ * popup window
+ * @param  {string|number} w
+ * @param  {string|number} h
+ * @return {{w, h, top, left}}       Popup window
+ */
+// source https://stackoverflow.com/a/16861050
+export function popupCenter(w, h) {
+  /* global screen */
+  // Fixes dual-screen position
+  //                      Most browsers      Firefox
+  var dualScreenLeft = window.screenLeft || screen.left
+  var dualScreenTop = window.screenTop || screen.top
+
+  var width = window.innerWidth
+    ? window.innerWidth
+    : document.documentElement.clientWidth
+    ? document.documentElement.clientWidth
+    : screen.width
+  var height = window.innerHeight
+    ? window.innerHeight
+    : document.documentElement.clientHeight
+    ? document.documentElement.clientHeight
+    : screen.height
+
+  var left = width / 2 - w / 2 + dualScreenLeft
+  var top = height / 2 - h / 2 + dualScreenTop
+  return {
+    w,
+    h,
+    top,
+    left
+  }
+}
+
+/**
+ * Renders a popup and listen to popup events
+ */
+export class Popup extends PureComponent {
+  constructor(props, context) {
+    super(props, context)
+
+    // url is not expected to change
+    const { initialUrl } = props
+    this.state = {
+      url: initialUrl
+    }
+
+    this.handleClose = this.handleClose.bind(this)
+    this.handleMessage = this.handleMessage.bind(this)
+    this.handleLoadStart = this.handleLoadStart.bind(this)
+  }
+
+  componentDidMount() {
+    this.showPopup()
+  }
+
+  componentWillUnmount() {
+    this.killPopup()
+  }
+
+  addListeners(popup) {
+    // Listen here for message FROM popup
+    window.addEventListener('message', this.handleMessage)
+
+    // rest of instructions only on mobile app
+    if (!isMobileApp()) return
```

Comment:
Nit: I think if we invert this condition, the code is more self-explanatory and the comment can be removed

```js
if (isMobileApp) {
  popup.addEventListener('loadstart', this.handleLoadStart)
  popup.addEventListener('exit', this.handleClose)
}
```

Reply:
I love early returns but I must admit that you're right on this point :)